src/bp/ui-shared/src/FormFields/TagInputList/SingleTag.tsx

```diff
@@ -0,0 +1,62 @@
+import { Tag, TagInput } from '@blueprintjs/core'
+import React, { FC, useEffect, useRef } from 'react'
+import { lang } from '~/translations'
+
+import { Item } from '.'
+import style from './style.scss'
+
+interface Props {
+  item: Item
+  placeholder: string
+  isFocused?: boolean
+  onChange: (item: Item) => void
+  onKeyDown: (event, canDelete: boolean) => void
+  onBlur: () => void
+}
+
+export const SingleTag: FC<Props> = ({ item, isFocused, placeholder, onChange, onKeyDown, onBlur }) => {
+  const inputRef = useRef<any>(null)
+  const canRemoveLast = useRef<boolean>(item.synonyms.length === 0)
+
+  useEffect(() => {
+    if (item.synonyms.length > 0) {
```

Comment:
What are synonyms here ? can you find a more generic name for this. And, quite frankly I got mixed up why a SingleTag (component name) can have multiple synonyms ..

Reply:
Yeah, i'll rename those. Wanted to avoid converting from tags to synonyms, but i'll do it correctly :P